assets/src/hooks/useLadderSvgHeight.ts

```diff
@@ -0,0 +1,35 @@
+import { Dispatch, RefObject, SetStateAction, useLayoutEffect } from "react"
+
+const useLadderSvgHeight = (
+  ref: RefObject<SVGSVGElement>,
+  x: number,
+  y: number,
+  width: number,
+  height: number,
+  setHeight: Dispatch<SetStateAction<number>>
+) => {
+  useLayoutEffect(() => {
+    if (
+      ref.current !== null &&
+      // only if there is no viewBox attribute
+      !ref.current.getAttribute("viewBox") &&
+      // only if rendered
+      ref.current.getBBox().width &&
+      ref.current.getBBox().height
+    ) {
+      const routeLadder: HTMLElement = document.getElementsByClassName(
+        "m-ladder"
+      )[0] as HTMLElement
+      const newHeight = routeLadder.offsetHeight
+      if (newHeight !== height) {
+        setHeight(newHeight)
+        ref.current.setAttribute("width", `${width}`)
+        ref.current.setAttribute("height", `${newHeight}`)
+        // (0, 0) is in the center of the first timepoint
+        ref.current.setAttribute("viewBox", [x, y, width, height].join(" "))
```

Comment:
Instead of having to reach into the DOM to mutate these, could they be constructed the normal way in `ladder.tsx` based on the mutated `height` value?

Reply:
It doesn't use the new height if you do it that way. I think it is rendering the svg before the hook is run.

Reply:
If it's using the height in the calculations inside the svg, it should also be able to use the height in the viewbox, right? Calling `setHeight` should trigger a rerender.

If the `setHeight` can trigger the rerender, then I think we could get rid of the `ref`.

Reply:
Try it, it doesn't work.

Reply:
Got it to work: https://github.com/mbta/skate/commit/be85eca9051538725216e38a675d84754f02c9cc

Reply:
Thanks! I pulled that in and updated the tests.